subsys/net/lib/mcumgr_smp_client/Kconfig

```diff
@@ -0,0 +1,35 @@
+#
+# Copyright (c) 2023 Nordic Semiconductor
+#
+# SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+#
+
+menuconfig NRF_MCUMGR_SMP_CLIENT
+	bool "MCUMGR SMP client library"
+	select BASE64
+	select DFU_TARGET
+	select DFU_TARGET_SMP
+	select DOWNLOAD_CLIENT
+	select FOTA_DOWNLOAD
+	select NET_BUF
+	select ZCBOR
```

Comment:
I'm not really too fond of this either. To put this into perspective, someone enables this, they have been forced to have base64 and CRC enabled, they can't disable them, but why are they being enabled when none of the other options need it? The option that actually does need it is not related to this, which would be the UART/shell MCUmgr transports